cmd/kubeadm/app/cmd/platform/init_unix.go

```diff
@@ -0,0 +1,95 @@
+// +build !windows
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package platform
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/user"
+	"strconv"
+)
+
+// InitCopyCredentialsForUser will copy the cluster admin credentials to the home path of a non-root user
+func InitCopyCredentialsForUser(copyCredentialsForUser string, adminKubeConfigPath string) error {
+
+	var err error
+	var uid, gid int
+	var usr *user.User
+	var src, dest *os.File
+	var kubeDir, configFilePath string
+
+	usr, err = user.Lookup(copyCredentialsForUser)
+	if err != nil {
+		return fmt.Errorf(InitErrorNoSuchUser)
```

Comment:
I'm not a fan of recasting the errors here.  Could you use the errors.Wrap utility if you want to put your output change on it.

Reply:
ok, i can rebase this next Monday (AFK this week).

Reply:
updated to always include the relevant `error` in the `Errorf()` format.